### ISSUE TITLE
Log graph does not update on push, pull, reset...

### ISSUE THREAD

**Issue by gollth** [NeogitOrg/neogit#938]
### Description

When I do some git action like pushing, fetching, rebasing or resetting, from the log graph view, the changes in the history are not reflected immediately. I have to `q` out of the log graph view and then `l b` back into the view to see the changes. `<C-r>` does not help either.

Ideally the log graph automatically updates, similar to how magit is doing it:

| Neogit | Magit |
|:---|:---|
|![neogit-push](https://github.com/NeogitOrg/neogit/assets/31999281/6cd33dfc-428f-4478-a705-83a5bcf0a4e6)|![magit-push](https://github.com/NeogitOrg/neogit/assets/31999281/2ef6ecda-6656-4f85-9733-08a1ac302326)|

### Neovim version

NVIM v0.10.0-dev
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Run "nvim -V1 -v" for more info

### Operating system and version

Ubuntu 20.04

### Steps to reproduce

1. Open Neogit 
2. Do any git action, which changes the log graqh, e.g. pushing or resetting
3. Note how the log graph is not updating

### Expected behavior

Similar behaviour to Magit, i.e. that the log graph automatically reflects changes in the history (or at least an easy way to refresh the log graph view, e.g. via autocmds...)

### Actual behavior

Log graph is not updating after a git action is performed

### Minimal config

```Lua
-- NOTE: See the end of this file if you are reporting an issue, etc. Ignore all the "scary" functions up top, those are
-- used for setup and other operations.
local M = {}

local base_root_path = vim.fn.fnamemodify(debug.getinfo(1, "S").source:sub(2), ":p:h") .. "/.min"
function M.root(path)
  return base_root_path .. "/" .. (path or "")
end

function M.load_plugin(plugin_name, plugin_url)
  local package_root = M.root("plugins/")
  local install_destination = package_root .. plugin_name
  vim.opt.runtimepath:append(install_destination)

  if not vim.loop.fs_stat(package_root) then
    vim.fn.mkdir(package_root, "p")
  end

  if not vim.loop.fs_stat(install_destination) then
    print(string.format("> Downloading plugin '%s' to '%s'", plugin_name, install_destination))
    vim.fn.system({
      "git",
      "clone",
      "--depth=1",
      plugin_url,
      install_destination,
    })
    if vim.v.shell_error > 0 then
      error(string.format("> Failed to clone plugin: '%s' in '%s'!", plugin_name, install_destination),
        vim.log.levels.ERROR)
    end
  end
end

---@alias PluginName string The plugin name, will be used as part of the git clone destination
---@alias PluginUrl string The git url at which a plugin is located, can be a path. See https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols for details
---@alias MinPlugins table<PluginName, PluginUrl>

---Do the initial setup. Downloads plugins, ensures the minimal init does not pollute the filesystem by keeping
---everything self contained to the CWD of the minimal init file. Run prior to running tests, reproducing issues, etc.
---@param plugins? table<PluginName, PluginUrl>
function M.setup(plugins)
  vim.opt.packpath = {}                      -- Empty the package path so we use only the plugins specified
  vim.opt.runtimepath:append(M.root(".min")) -- Ensure the runtime detects the root min dir

  -- Install required plugins
  if plugins ~= nil then
    for plugin_name, plugin_url in pairs(plugins) do
      M.load_plugin(plugin_name, plugin_url)
    end
  end

  vim.env.XDG_CONFIG_HOME = M.root("xdg/config")
  vim.env.XDG_DATA_HOME = M.root("xdg/data")
  vim.env.XDG_STATE_HOME = M.root("xdg/state")
  vim.env.XDG_CACHE_HOME = M.root("xdg/cache")

  -- NOTE: Cleanup the xdg cache on exit so new runs of the minimal init doesn't share any previous state, e.g. shada
  vim.api.nvim_create_autocmd("VimLeave", {
    callback = function()
      vim.fn.system({
        "rm",
        "-r",
        "-f",
        M.root("xdg")
      })
    end
  })
end

-- NOTE: If you have additional plugins you need to install to reproduce your issue, include them in the plugins
-- table within the setup call below.
M.setup({
  plenary = "https://github.com/nvim-lua/plenary.nvim.git",
  telescope = "https://github.com/nvim-telescope/telescope.nvim",
  diffview = "https://github.com/sindrets/diffview.nvim",
  neogit = "https://github.com/NeogitOrg/neogit"
})
-- WARN: Do all plugin setup, test runs, reproductions, etc. AFTER calling setup with a list of plugins!
-- Basically, do all that stuff AFTER this line.
require("neogit").setup({}) -- For instance, setup Neogit
```

**Comment by gollth** [comment]
Hmm maybe something like `autocmd` is possible?

```lua
local group = vim.api.nvim_create_augroup("MyCustomNeogitEvents", { clear = true })
vim.api.nvim_create_autocmd("User", {
  pattern = "NeogitStatusRefreshed",
  group = group,
  callback = function()
    local buffername = vim.fn.expand("%:t")
    if buffername == "NeogitLogView" then
      -- What to call here?!
    end
  end,
})
```

**Comment by gollth** [comment]
Solved by #981

**Comment by jonpalmisc** [comment]
This has the caveat that the type of log view to open on "refresh" is hardcoded (to "b"/branches in this case). This means that you might get a different view on refresh than you had opened originally.

Would still be interested to see this supported out of the box, like in Magit.
